internal/advertisement-operator/controller.go

```diff
@@ -163,6 +169,11 @@ func (r *AdvertisementReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
+// checks that the Advertisement is being deleted
+func (r *AdvertisementReconciler) isDeleting(adv *advtypes.Advertisement) bool {
+	return adv.DeletionTimestamp != nil
```

Comment:
The `DeletionTimestamp` field could be set to `zero`, to be sure we check that case we could use `adv.DeletionTimestamp.IsZero()` method, which returns true if the value is nil or time is zero.